hota/Mods/Artifacts/Content/config/hota/artifacts/goldenGoose.json

```diff
@@ -1,23 +1,31 @@
 {
 	"goldenGoose" : {
+		"type" : [ "HERO" ],
+		"slot" : "MISC",
+		"class" : "RELIC",
+		"value" : 0,
+		"components" : [
+			"endlessPurseOfGold",
+			"endlessBagOfGold",
+			"endlessSackOfGold"
+		],
 		"bonuses" : [
 			{
-				"subtype" : "resource.gold",
 				"type" : "GENERATE_RESOURCE",
-				"val" : 4750,
-				"valueType" : "BASE_NUMBER"
+				"subtype" : "resource.gold",
+				"valueType" : "BASE_NUMBER",
+				"val" : 7000
```

Comment:
Recheck I guess but old version should be correct:
- In addition to component effects brings 4750 Gold gold/day, for a total bonus of 7000 Gold gold/day.
- Component effects: bring 2250 Gold gold/day.

So, artifact *itself* give 4750 gold. But due to component effects, total income is 7000

Reply:
Maybe I do not understand something...
This is not 'new artifact' that replaces completely components and just blocks some slots? It performs additive calculations for components?

Reply:
Yep. Bonuses of individual components are still active.
Normally component and composite effects are different.
For example Angelic Alliance only defines alignment mixing and prayer - bonuses to primary skills still come from components:
https://github.com/vcmi/vcmi/blob/develop/config/artifacts.json#L1919
Golden Goose is a bit special here since it has same effect (gold production) as its components

Reply:
Ok will fix that.
Sounds strange this approach/limitation... to calculate hidden in the background, individual bonuses from components. If provided as a shortcut/option... sure, but not default.
I also wonder about readability/extensibility of the code implementing this...

In this case, I as a modder do not care about those arithmetics. I care that golden goose gives 7000g. Which is simple and straight forward.

Not to mention that this should be a fused artifact, in the end. No  more components no nothing, just the goose :)

Reply:
You do have a point, but this is how all H3 composite artifacts work - they only extend effect.
H3 tooltips also only state effect from composite artifact. Effects from individual component are omitted from description.

It is indeed less confusing than Golden Goose - since H3 composites provide effects different from their components, so no need for arithmetic.

Reply:
ok reverted and proposed one sentence for clarification. If ok than this could be merge from my side.